apps/andi/test/integration/andi_web/live/data_dictionary_form_test.exs

```diff
@@ -180,6 +180,145 @@ defmodule AndiWeb.DataDictionaryFormTest do
+      assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
+      data_dictionary_view = find_child(view, "data_dictionary_form_editor")
+
+      html = render_hook(data_dictionary_view, "file_upload", %{"fileSize" => 200_000_001})
+
+      refute Enum.empty?(find_elements(html, "#schema_sample-error-msg"))
+    end
+
+    test "provides modal when existing schema will be overwritten", %{conn: conn} do
+      dataset = TDG.create_dataset(%{technical: %{sourceFormat: "text/csv"}})
+
+      {:ok, _} = Datasets.update(dataset)
+      assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
+      data_dictionary_view = find_child(view, "data_dictionary_form_editor")
+
+      csv_sample = "CAM\nrules"
+
+      html = render_hook(data_dictionary_view, "file_upload", %{"fileSize" => 100, "fileType" => "text/csv", "file" => csv_sample})
+
+      refute Enum.empty?(find_elements(html, ".overwrite-schema-modal--visible"))
+    end
+
+    test "does not provide modal with no existing schema", %{conn: conn} do
+      dataset =
+        TDG.create_dataset(%{technical: %{sourceType: "remote", sourceFormat: "text/csv"}})
+        |> Map.update(:technical, %{}, &Map.delete(&1, :schema))
+
+      {:ok, _} = Datasets.update(dataset)
+      assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
+      data_dictionary_view = find_child(view, "data_dictionary_form_editor")
+
+      csv_sample = "CAM\nrules"
+
+      html = render_hook(data_dictionary_view, "file_upload", %{"fileSize" => 100, "fileType" => "text/csv", "file" => csv_sample})
+
+      assert Enum.empty?(find_elements(html, ".overwrite-schema-modal--visible"))
+
+      updated_dataset = Datasets.get(dataset.id)
+      refute Enum.empty?(updated_dataset.technical.schema)
+    end
+
+    test "parses CSVs with various types", %{conn: conn} do
+      dataset =
+        TDG.create_dataset(%{technical: %{sourceType: "remote", sourceFormat: "text/csv"}})
+        |> Map.update(:technical, %{}, &Map.delete(&1, :schema))
+
+      {:ok, _} = Datasets.update(dataset)
+      assert {:ok, view, html} = live(conn, @url_path <> dataset.id)
+      data_dictionary_view = find_child(view, "data_dictionary_form_editor")
+
+      csv_sample = "string,int,float,bool,date\nabc,9,1.5,true,2020-07-22T21:24:40"
+
+      render_hook(data_dictionary_view, "file_upload", %{"fileSize" => 100, "fileType" => "text/csv", "file" => csv_sample})
+
+      updated_dataset = Datasets.get(dataset.id)
+
+      generated_schema =
+        updated_dataset.technical.schema
+        |> Enum.map(fn item -> %{type: item.type, name: item.name} end)
+
+      expected_schema = [
+        %{name: "bool", type: "boolean"},
+        %{name: "date", type: "date"},
+        %{name: "float", type: "float"},
+        %{name: "int", type: "integer"},
+        %{name: "string", type: "string"}
+      ]
+
+      assert generated_schema == expected_schema
+    end
+
+    test "handles invalid json", %{conn: conn} do
```

Comment:
Is it possible for CSV to be invalid too?

Also, are we gracefully handling errors reading/uploading the file. Not easy to test, but worth at least manually verifying if possible.

Reply:
Since the file is coming back as a string, I'm not sure that a CSV can be invalid. But I might be wrong about that.

Reply:
We may need to add something to the hook to send back errors reading the file. I will look into that

Reply:
It looks like errors reading the file will get caught by the event handler that adds a generic error message, so as far as I can tell, we are handling these errors